server/mdm/apple/apple_mdm.go

```diff
@@ -288,18 +352,112 @@ func NewDEPService(
 				level.Info(kitlog.With(logger)).Log("msg", "no DEP hosts to add")
 			}
 
-			// TODO(mna): at this point, the hosts rows are created for the devices, with the
+			// at this point, the hosts rows are created for the devices, with the
 			// correct team_id, so we know what team-specific profile needs to be applied.
-			return assigner.ProcessDeviceResponse(ctx, resp)
+			return depSvc.processDeviceResponse(ctx, depClient, resp, teamID)
 		}),
 	)
 
-	return &DEPService{
-		syncer:     syncer,
-		depStorage: depStorage,
-		logger:     logger,
-		ds:         ds,
+	return depSvc
+}
+
+// processDeviceResponse processes the device response from the device sync
+// DEP API endpoints and assigns the profile UUID associated with the DEP
+// client DEP name.
+func (d *DEPService) processDeviceResponse(ctx context.Context, depClient *godep.Client, resp *godep.DeviceResponse, tmID *uint) error {
```

Comment:
This is copied and adapted from nanomdm, as we need to get the profile UUID from either the default profile's nano store, or the setup assistant table.